asapdiscovery-ml/asapdiscovery/ml/cli.py

```diff
@@ -0,0 +1,467 @@
+import json
+from pathlib import Path
+
+import asapdiscovery.ml.schema_v2.config as ascfg
+import click
+
+
+@click.group()
+def ml():
+    pass
+
+
+@ml.command()
+@click.option(
+    "-o",
+    "--output-dir",
+    required=True,
+    type=click.Path(
+        exists=False, file_okay=False, dir_okay=True, writable=True, path_type=Path
+    ),
+    help=(
+        "Top-level output directory. A subdirectory with the current W&B "
+        "run ID will be made/searched if W&B is being used."
+    ),
+)
+# Model setup args
+@click.option(
+    "-model",
+    "--model-type",
+    required=True,
+    type=ascfg.ModelType,
+    help="Which model type to use.",
+)
+@click.option(
+    "--config-file",
+    type=click.Path(exists=True, dir_okay=False, path_type=Path),
+    help=(
+        "JSON file giving model config. Any passed CLI args will overwrite the options "
+        "in this file."
+    ),
+)
+# W&B args
+@click.option("--use-wandb", is_flag=True, help="Use W&B to log model training.")
+@click.option("--sweep", is_flag=True, help="This run is part of a W&B sweep.")
+@click.option("-proj", "--wandb-project", help="W&B project name.")
+@click.option("-name", "--wandb-name", help="W&B project name.")
+@click.option(
+    "-e",
+    "--extra_config",
+    multiple=True,
+    help=(
+        "Any extra config options to log to W&B, provided as comma-separated pairs. "
+        "Can be provided as many times as desired "
+        "(eg -e key1,val1 -e key2,val2 -e key3,val3)."
+    ),
+)
+# Shared MTENN-related parameters
+@click.option(
+    "--grouped",
+    type=bool,
+    help="Model is a grouped (multi-pose) model.",
+)
+@click.option(
```

Comment:
If you plan on re-using these decorators across multiple scripts you can do something like 

```
def blah(func):
    return click.option(
        "--blah",
        is_flag=True,
        default=False,
        help="blah",
    )(func)
````

and the use like `@blah` but up to you

Reply:
I think I'll leave them as is for now, but might refactor into functions when I start building out other ML scripts

Reply:
sounds good.